core/dbt/task/compile.py

```diff
@@ -85,4 +86,4 @@ def defer_to_manifest(self, adapter, selected_uids: AbstractSet[str]):
             selected=selected_uids,
         )
         # TODO: is it wrong to write the manifest here? I think it's right...
-        self.write_manifest()
+        write_manifest(self.manifest, self.config.target_path)
```

Comment:
I remember in changes above we write manifest in the click command, any reason we do this here not following the same convention?

Reply:
This is to maintain parity with existing functionality. I am not well versed in why writing the manifest happens when it does, however I didn't want to change anything currently happening, only abstract manifest generation.